### ISSUE TITLE
Invalid expression type with set of asyncio tasks

### ISSUE THREAD

**Issue by decaz** [python/mypy#9111]
Listing:
```python
import asyncio
from typing import NoReturn, Set

tasks: Set[asyncio.Task[NoReturn]] = set()
```
mypy error:
```
$ mypy xxx.py 
xxx.py:4: error: Incompatible types in assignment (expression has type "Set[<nothing>]", variable has type "Set[Task[NoReturn]]")
Found 1 error in 1 file (checked 1 source file)
```
Environment:
```
$ python -V
Python 3.8.3
$ mypy -V
mypy 0.782
```

**Comment by Akuli** [comment]
also happens with lists and dicts, but doesn't happen if there's e.g. `str` instead of `NoReturn`

**Comment by JukkaL** [comment]
The issue is with `NoReturn`, not with `asyncio.Task`. This also generates the error:

```py
from typing import NoReturn, List

x: List[List[NoReturn]] = []  # Unexpected error
```

**Comment by ilevkivskyi** [comment]
This now works correctly.
